Documentation/articles/command/command_handler.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Handling Commands
+description: Recommendations and Requirements when Handling a Command
+keywords: 
+author: einari, smithmx
+---
+
+# Command Handler
+
+## Background
+
+A *Command Handler* handles commands.  The command handler is responsible for executing the appropriate methods on Aggregate Roots and Domain Services to fulfill the intent of the *Command*.
+
+## Requirements
+
+A *Command Handler* must implement the *IHandleCommands* interface.  This is a simple marker interface with no methods to implement.
+
+To handle a *Command*, you create a method called *Handle* with a single parameter that is the *Command* you wish to handle.  A *Command Handler* does not return anything.  It is not permitted to return a value from a *Command*.  When a *Command* is executed, you will receive a *CommandResult*
+which will indicate whether the *Command* succeeded and, in the case of failure, why the *Command* failed.  No artifacts that were created or updated as a result of the *Command* will be present on the *CommandResult*.
+
+```csharp
+public void Handle(AddRecommendationToCart cmd)
+```
+
+There **MUST** be a single handler for a Command.  Dolittle will throw an exception if there is no handler for a command or if there is more than one handler for a command.  There can only be a single handler for a command as there must be a clear and unambiguous indication of the result of the execution of the command.  Multiple command handlers for a single command would require co-ordination of multiple results with associated rollback of successful handlers after unsuccessful ones.  Where you wish multiple things to happen on receipt of a Command, you must implement and handle this yourself in your *Command Handler*.
+
```

Comment:
We'll probably want to re-inforce this by referring to the (unwritten) Event Store documentation that says that a commit is against a single event source.  Indeed, the whole thing is about a transaction.  A command is a transaction.  Therefore the command handler must be a transaction.  Since a transaction can only be a single commit and the Event Store only supports a commit against a single event source, then a command handler can only get a single event source.

This is more just a note to myself that we need to flesh this out a bit later! :-)